pkg/daemon/gateway.go

```diff
@@ -106,6 +106,88 @@ func (c *Controller) setIPSet() error {
 	return nil
 }
 
+func (c *Controller) addIPSetMembers(setID, subnet, ip string) {
+	podSubnet, err := c.subnetsLister.Get(subnet)
+	if err != nil {
+		klog.Errorf("get subnet %s failed, %+v", subnet, err)
+		return
+	}
+
+	if !podSubnet.Spec.NatOutgoing ||
+		podSubnet.Spec.Vpc != util.DefaultVpc ||
+		podSubnet.Spec.GatewayType != kubeovnv1.GWDistributedType {
+		return
+	}
+
+	protocols := make([]string, 2)
+	if c.protocol == kubeovnv1.ProtocolDual {
+		protocols[0] = kubeovnv1.ProtocolIPv4
+		protocols[1] = kubeovnv1.ProtocolIPv6
+	} else {
+		protocols[0] = c.protocol
+	}
+
+	podIPs := strings.Split(ip, ",")
```

Comment:
Check ip protocol directly and add it to related ipset, no need to create `protocols` array